modular_skyrat/master_files/code/modules/clothing/clothing.dm

```diff
@@ -12,14 +12,18 @@
     return ..()
 
 // Generates a coloured species-specific variant of the clothing item
-/obj/item/clothing/proc/generate_species_clothing(file_to_use, state_to_use, species) 
-	var/icon/human_clothing_icon = icon(file_to_use, state_to_use)
+/obj/item/clothing/proc/generate_species_clothing(file_to_use, state_to_use, species, default_file_to_use)
+	var/list/clothing_states = icon_states(file_to_use)
 
-	if(!species_clothing_colors || !species_clothing_icon_state)
-		// This clothing piece has no species specific clothing, use the normal icon
+	var/icon/human_clothing_icon = icon(file_to_use, state_to_use) //overriden clothing icon
+	var/icon/default_clothing_icon = icon(default_file_to_use, state_to_use) //non overriden clothing icon, important for getting the color matrix to not die
```

Comment:
This is called only once per clothing item, *right?* Because otherwise I can see this being quite expensive.

Reply:
yeah, it should be called only once per item